samples/csharp_dotnetcore/80.skills-simple-bot-to-bot/SimpleRootBot/AdapterWithErrorHandler.cs

```diff
@@ -17,57 +18,96 @@ namespace Microsoft.BotBuilderSamples.SimpleRootBot
-                            endOfConversation.ApplyConversationReference(turnContext.Activity.GetConversationReference(), true);
+        protected async Task sendErrorMessageAsync(ITurnContext turnContext, Exception exception)
+        {
+            // Send a message to the user
+            var errorMessageText = "The bot encountered an error or bug.";
+            var errorMessage = MessageFactory.Text(errorMessageText, errorMessageText, InputHints.IgnoringInput);
+            await turnContext.SendActivityAsync(errorMessage);
 
-                            await skillClient.PostActivityAsync(botId, activeSkill, skillsConfig.SkillHostEndpoint, (Activity)endOfConversation, CancellationToken.None);
-                        }
+            errorMessageText = "To continue to run this bot, please fix the bot source code.";
+            errorMessage = MessageFactory.Text(errorMessageText, errorMessageText, InputHints.ExpectingInput);
+            await turnContext.SendActivityAsync(errorMessage);
 
-                        // Delete the conversationState for the current conversation to prevent the
-                        // bot from getting stuck in a error-loop caused by being in a bad state.
-                        // ConversationState should be thought of as similar to "cookie-state" in a Web pages.
-                        await conversationState.DeleteAsync(turnContext);
-                    }
-                    catch (Exception ex)
-                    {
-                        logger.LogError(ex, $"Exception caught on attempting to Delete ConversationState : {ex}");
-                    }
-                }
+            // Send a trace activity, which will be displayed in the Bot Framework Emulator
+            await turnContext.TraceActivityAsync("OnTurnError Trace", exception.ToString(), "https://www.botframework.com/schemas/error", "TurnError");
+        }
+
+        private async Task<InvokeResponse> endSkillConversationAsync(ITurnContext turnContext)
+        {
+            if (_conversationState == null || _skillClient == null || _skillsConfig == null)
+            {
+                return null;
+            }
 
-                // Send a trace activity, which will be displayed in the Bot Framework Emulator
-                await turnContext.TraceActivityAsync("OnTurnError Trace", exception.ToString(), "https://www.botframework.com/schemas/error", "TurnError");
-            };
+            try
+            {
+                await _conversationState.SaveChangesAsync(turnContext, true);
+
+                // Inform the active skill that the conversation is ended so that it has
+                // a chance to clean up.
+                // Note: "activeSkillProperty" is set by the RooBot while messages are being
+                // forwarded to a Skill.
+                var activeSkill = await _conversationState.CreateProperty<BotFrameworkSkill>("activeSkillProperty").GetAsync(turnContext, () => null);
```

Comment:
Consider making "activeSkillProperty" a public const string in RootBot and update the code there too to use it.
It is still not awesome but at least we don't have the string hardcoded in two places.